### PR TITLE
check txz/zip counterparts also with in-use as with deletion

### PR DESCRIPTION
**Pull request by partouf**
* checks during `builds_rm_old()` if a version has a windows/linux counterpart in-use (the list of releases lists .zip and .txz separately even if they are the same gh-versionnumber, so it might think something is not in use while it really is)
* if sure to remove, in `remove_release()` also remove the .zip files if they are there

Had to rewrite the delete magic in https://github.com/compiler-explorer/infra/compare/fix-old-build-deletions?expand=1#diff-98b39f6ed2a45cf3cd6cddfb72637ddf89584cf5af3dc21eba56d47c3ade26d4R126 - I think it's correct, but could use another eye on it

### CODE REVIEW

**Comment by partouf** [comment]
Fixes #947

File: bin/lib/cli/builds.py [hunk 1]
```diff
@@ -170,8 +171,11 @@ def builds_rm_old(dry_run: bool, max_age: int):
     for release in get_all_releases():
         max_builds[release.version.source] = max(release.version.number, max_builds[release.version.source])
     for release in get_all_releases():
-        if release.key in current:
+        if (release.key in current) or (get_key_counterpart(release.key) in current):
```
Comment (mattgodbolt) [inline bin/lib/cli/builds.py:174]: I'd also extract `counterpart_key` as a variable as you're using `get_key_counterpart(release.key)` in three places in the 4 or so lines here.

File: bin/lib/cli/builds.py [hunk 2]
```diff
@@ -170,8 +171,11 @@ def builds_rm_old(dry_run: bool, max_age: int):
...
             print(f"Skipping {release} as it is a current version")
+            if dry_run:
+                if get_key_counterpart(release.key) in current:
```
Comment (mattgodbolt) [inline bin/lib/cli/builds.py:177]: Super useful for debug: thanks!

**Review by mattgodbolt — APPROVED** [review]
Some comments on how to improve...nothing blocking though. general approach seems ok, thanks!

**Review by mattgodbolt — APPROVED** [review]
Some minor suggestions, but basically lgtm: thanks!

**Review by mattgodbolt — APPROVED** [review]
Some minor suggestions, but basically lgtm: thanks!

**Comment by mattgodbolt** [comment]
trying to submit on a flakey internet connection, sorry for the reposts

File: bin/lib/amazon.py [hunk 3]
```diff
@@ -92,12 +92,38 @@ def get_autoscaling_groups_for(cfg: Config) -> List[dict]:
     return result
 
 
+def s3_file_exists(key: str) -> bool:
+    try:
+        s3_client.get_object(Bucket="compiler-explorer", Key=key)
```
[the remark this answers is not in the document]
Comment (partouf) [reply bin/lib/amazon.py:97]: Ah excellent, will do

File: bin/lib/cli/builds.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (partouf) [reply bin/lib/cli/builds.py:174]: I did try but got a strange error, but will try again

Comment (partouf) [reply bin/lib/cli/builds.py:174]: good point

[the remark this answers is not in the document]
Comment (partouf) [reply bin/lib/cli/builds.py]: seems to work indeed

**Pull request merged**